bluetooth: CS various updates

olivier-le-sage opened the pull request:
Various channel sounding-related changes/fixes

olivier-le-sage committed:
bluetooth: host: Use LOG_WRN for failed CS procedures

These can indicate that something about the user configuration needs to
be changed.

Signed-off-by: Olivier Lesage <olivier.lesage@nordicsemi.no>

olivier-le-sage committed:
bluetooth: host: document that step_data_buf can be NULL

If the controller reports 0 steps in the subevent results, the pointer
is NULL.

Signed-off-by: Olivier Lesage <olivier.lesage@nordicsemi.no>

olivier-le-sage committed:
bluetooth: samples: Wait for encryption before starting CS setup

Handling the security changed callback was missing, so it could happen
that the sample would try to read remote capabilities before the link
was encrypted, which is not allowed.

Signed-off-by: Olivier Lesage <olivier.lesage@nordicsemi.no>

olivier-le-sage committed:
bluetooth: samples: Remove auto phy update from connected CS samples

Unless the PHY is managed by LE power control, the CS procedure phy
needs to match the ACL phy.

Disabling the auto phy updates keeps the ACL on 1M.

The alternative would be to read the current phy by enabling
BT_USER_PHY_UPDATE but this still doesn't work if the phy is changed
after being read.

Signed-off-by: Olivier Lesage <olivier.lesage@nordicsemi.no>

sean-madigan commented on include/zephyr/bluetooth/hci_types.h:
```diff
@@ -3689,8 +3689,8 @@ struct bt_hci_le_cs_step_data_mode_1 {
 	union {
-		int16_t toa_tod_initiator;
-		int16_t tod_toa_reflector;
+		uint16_t toa_tod_initiator;
```
Is it worth some extra documentation with these?

alwa-nordic commented:
>  bluetooth: host: change CS ToX_ToX timestamps back to uint16_t
> 
> The special value 0x8000 is just outside the range of an int16_t
so to avoid potential signed overflow behavior issues, casting this
value is made the user's responsibility

Can you explain the signed overflow behavior issues?

jhedberg commented:
> BT_HCI_LE_CS_TIME_DIFFERENCE_NOT_AVAILABLE is bigger than INT16_MAX. That's all really

Would it make sense to define the macro as `((int16_t)0x8000)'?

olivier-le-sage commented:
> > BT_HCI_LE_CS_TIME_DIFFERENCE_NOT_AVAILABLE is bigger than INT16_MAX. That's all really
> 
> Would it make sense to define the macro as `((int16_t)0x8000)`?

instead of this change from int16_t to uint16_t, right ? I guess that's fine

jhedberg commented:
> > Would it make sense to define the macro as `((int16_t)0x8000)`?
> 
> instead of this change from int16_t to uint16_t, right ? I guess that's fine

Yes, that's what I meant. That'd be more in line with how the HCI spec defines this parameter (it states that it's a signed integer).

olivier-le-sage committed:
bluetooth: host: cast LE_CS_TIME_DIFFERENCE_NOT_AVAILABLE explicitly

The special value 0x8000 is just outside the range of an int16_t,
so this should help make sure whatever overflow behavior is entered
by using the step data structs on protocol bytes is the same as this
macro.

Signed-off-by: Olivier Lesage <olivier.lesage@nordicsemi.no>
